src/Services/Core/MemberService.php

```diff
@@ -233,4 +246,105 @@ public function getLatestBoard(): array
         return $boardMembers;
 
     }
+
+    /**
+     * @param array $fields
+     * @return Member
+     * @throws \Exception
+     */
+    public function createPhoto(int $id, ?array $fields): String
+    {
+        if ($fields['file'] == null) {
+            $msg = 'File is empty in given parameters';
+            $this->logger->error($msg);
+            throw new KerosException($msg, 400);
+        }
+        $file = $fields['file'];
+
+        $id = Validator::requiredId($id);
+        $member = $this->getOne($id);
+
+        if (!$member) {
+            throw new KerosException("The member could not be found", 404);
+        }
+
+        $filename = $member->getProfilePicture();
+
+        if ($filename) {
+            $filepath =  $this->kerosConfig['MEMBER_PHOTO_DIRECTORY'] . $filename;
+            if (file_exists($filepath)){
+                unlink($filepath);
+            }
+        }
+        
+        $filename = $this->directoryManager->uniqueFilename($file, false, '');
+
+        $this->directoryManager->mkdir($this->kerosConfig['MEMBER_PHOTO_DIRECTORY'] . pathinfo($filename, PATHINFO_DIRNAME));
+        $member->setProfilePicture($filename);
+
+        $this->memberDataService->persist($member);
+
+        return $filename;
+    }
+
+    /**
+     * @param int $id
+     * @throws \Keros\Error\KerosException
+     */
+    public function deletePhoto(int $id): void
+    {
+        $id = Validator::requiredId($id);
+
+        $member = $this->getOne($id);
+
+        if (!$member) {
+            throw new KerosException("The member could not be found", 404);
+        }
+
+        $filename = $member->getProfilePicture();
+
+        if (!$filename) {
+            throw new KerosException("Profile picture could not be found", 404);
+        }
+
+        $filepath =  $this->kerosConfig['MEMBER_PHOTO_DIRECTORY'] . $filename;
+
+        if (file_exists($filepath)){
+            unlink($filepath);

```

Comment:
Wow, who knew that "unlink" actually deletes a file :)

Reply:
Yup ^^